libs/utils/src/lib/utils.ts

```diff
@@ -24,11 +24,50 @@ export function validateEnvVars<Values extends ObjectShape>(
   try {
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     return validateSchema(schema, process.env as any);
-  } catch (err) {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  } catch (err: any) {
     const error: yup.ValidationError = err;
 
     throw new Error(
       `Invalid environment variables: ${error.message}. Make sure they are properly set in a .env or .env.local file`
     );
   }
 }
+
+export enum MetaMaskErrorCodes {
+  UserRejection = 4001,
+  InvalidParameters = -32602,
+  InternalError = -32603,
+}
+
+interface ProviderRpcError extends Error {
+  message: string;
+  code: number;
+  data?: unknown;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function isMetaMaskError(error: any): error is ProviderRpcError {
+  if (error.code && error.message) {
+    return true;
+  }
+  return false;
+}
+
+export function handleMetaMaskError(error: ProviderRpcError) {
+  if (error.code === MetaMaskErrorCodes.UserRejection) {
+    return;
+  }
+
+  if (error.data && (error.data as { message?: string }).message) {
+    const revertErrorMessage = (
+      error.data as { message?: string }
+    ).message?.split('revert ')[1];
+
+    alert(revertErrorMessage);
+    return;
+  }
+
+  console.log(error);
```

Comment:
```suggestion
  console.error(error);
```

We should iterate more over this function when we figure out more about the architecture of the app. But just a little detail. When logging errors, use `console.error`